internal/pkg/chain/pubqueue.go

```diff
@@ -143,24 +185,30 @@ func doRefresh() {
 	}
 	publishQueueWatcher.running = true
 
-	// remove succeed trx (wasm compatable)
-	publishQueueWatcher.db.PrefixCondDelete([]byte(PUBQUEUE_PREFIX), func(k []byte, v []byte, err error) (bool, error) {
-		if err != nil {
-			chain_log.Warnf("<pubqueue>: %s", err.Error())
-			// continue
-			return false, nil
-		}
-		item, err := ParsePublishQueueItem(k, v)
-		if err != nil {
-			chain_log.Warnf("<pubqueue>: %s", err.Error())
-			return false, nil
-		}
-		if item.State == PublishQueueItemStateSuccess {
-			return true, nil
-		}
+	if autoAck {

```

Comment:
Add a timer / ticker to auto ack the queue every x minutes?

Reply:
@huo-ju  actually, it shares the same ticker with the checking process(10s).

here: https://github.com/chux0519/quorum-1/blob/619ed0667350f033ad1e10c51fbfe99218eff447/internal/pkg/chain/pubqueue.go#L176

it works OK now, I think there's no need to use antoher ticker.

But, using another ticker still make sense, since auto-ack could be viewed as a different task.

What do you think